snuba/optimize.py

```diff
@@ -128,19 +253,29 @@ def get_partitions_to_optimize(
     return parts
 
 
-def _subdivide_parts(
-    parts: Sequence[util.Part], number_of_subdivisions: int
-) -> Sequence[Sequence[util.Part]]:
+def _subdivide_partitions(
```

Comment:
I am a bit concerned that the code in this module is becoming quite hard to follow. I think it is because there are two concerns that are mixed together while an abstraction could help.
Specifically we mix the scheduling aspect and the execution aspect.

What about having an `OptimizeScheduler` class that has a method  `get_next_schedule`. This method returns a sequence of seuqences of partitions. Each sequence is the schedule for a thread and a cutoff time. Once the cutoff time is passed you get the next schedule (and provide the set of successfully executed partitions).

Reply:
That was a good suggestion. I have incorporated the changes. Could you review the code again and see if the changes are inline with what you had in mind.